addons/mrp/migrations/10.0.2.0/post-migration.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+# © 2017 Paul Catinean <https://www.pledra.com>
+# Copyright 2017 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def fill_defaults(env):
+    # Set default values from field definition as this is a new field and
+    # the default val resembles the previous behavior best
+    default_specs = {
+        'mrp.bom': [('ready_to_produce', None)],
+        'mrp.routing.workcenter': [('batch', None)],
+        'mrp.workcenter': [('sequence', None)],
+    }
+    openupgrade.set_defaults(env.cr, env, default_specs)
+
+
+def map_mrp_production_state(cr):
+    openupgrade.map_values(
+        cr,
+        openupgrade.get_legacy_name('state'), 'state',
+        [('draft', 'confirmed'),
+         ('in_production', 'progress'),
+         ('ready', 'planned'),
+         ], table='mrp_production', write='sql')
+
+
+def map_mrp_workorder_state(cr):
+    openupgrade.map_values(
+        cr,
+        openupgrade.get_legacy_name('state'), 'state',
+        [('draft', 'pending'),
+         ('starworking', 'progress'),
+         ('pause', 'ready'),
+         ], table='mrp_workorder', write='sql')
+
+
+def update_stock_warehouse(cr):
+    cr.execute(
```

Comment:
You are doing this by SQL, so no recomputation is triggered as you mention on the analysis file (`...so as to force the creation of the default picking types and pull rules for manufacturing on that warehouse`)